registry/client/repository.go

```diff
@@ -211,8 +211,6 @@ func (ms *manifests) Tags() ([]string, error) {
 		}
 
 		return tagsResponse.Tags, nil
-	} else if resp.StatusCode == http.StatusNotFound {
-		return nil, nil
```

Comment:
I think this will cause a behavioral change in the engine.

The current code reads like this:

                tags, err = manSvc.Tags()
                if err != nil {
                        return err
                }

Then there's a for-range loop over `tags` later in the function. So unless I'm missing something, a 404 result would cause the client not to error out before, but this change would make it abort the pull with an error. The changed behavior seems more correct to me, but let's make sure this is okay.